### ISSUE TITLE
Worried benchmarks don't re-shuffle the list between each benchmark iteration

### ISSUE THREAD

**Issue by sts10** [sts10/sort_bench#3]

**Comment by oylenshpeegul** [comment]
Use [iter_batched_ref](https://docs.rs/criterion/latest/criterion/struct.Bencher.html#method.iter_batched_ref) instead of `iter` maybe?

```rust
    group.bench_function("Using regular stable sort", |b| {
        b.iter_batched_ref(
            || unsorted_word_list.clone(),
            |mut uwl| stable_sort(&mut uwl, &locale),
            BatchSize::SmallInput,
        )
    });
```

**Comment by sts10** [comment]
Thanks! fixed in 8f14b2d
